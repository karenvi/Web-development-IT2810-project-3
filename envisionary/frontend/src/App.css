.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

html {
  font-size: 18px;
}

body {
  background-color: aliceblue;
}

/* Makes a visually hidden header title for placeholders, makes it possible for screen readers to read text */
/* (CSS by: https://www.emgoto.com/) */
.visually-hidden {
  clip: rect(0 0 0 0);
  clip-path: inset(50%);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}