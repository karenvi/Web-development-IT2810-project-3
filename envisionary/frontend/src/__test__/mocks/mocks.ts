
import { GET_COUNTRIES, GET_REVIEWS_BY_COUNTRY_NAME } from "../../graphql/queries";





export const mocks = [
    {
      request: {
        query: GET_COUNTRIES,
        variables: {}
      },
      result: {
        data: {
            country: 
            [{ _id: "1", Area: "120538", CCA3: "PRK", Capital: "Pyongyang", Continent: "Asia", Country: "North Korea", Density: "216.2755", 
            GrowthRate: "1.0038", Population1970: "14996879", Population1980: "17973650", Population1990: "20799523", Population2000: "23367059",
            Population2010: "24686435", Population2015: "25258015", Population2020: "25867467", Population2022: "26069416", Rank: "56", 
            WorldPopulationPercentage: "0.33" },

            { _id: "2", Area: "25713", CCA3: "MKD", Capital: "Skopje", Continent: "Europe", Country: "North Macedonia", Density: "81.4218", GrowthRate: "0.9954",
            Population1970: "1656783", Population1980: "1907023", Population1990: "2044174", Population2000: "2037936", Population2010: "2093828",
            Population2015: "2107962", Population2020: "2111072", Population2022: "2093599", Rank: "150", WorldPopulationPercentage: "0.03"},

            { _id: "3", Area: "464", CCA3: "NFK", Capital: "Saipan", Continent: "Oceania", Country: "Northern Mariana Islands", Density: "106.7909", 
            GrowthRate: "1.0014", Population1970: "10143", Population1980: "17613", Population1990: "48002", Population2000: "80338", Population2010: "54087",
            Population2015: "51514", Population2020: "49587", Population2022: "49551", Rank: "210", WorldPopulationPercentage: "0"},
            ]
        }
      }
    },
    {
      request: {
        query: GET_REVIEWS_BY_COUNTRY_NAME,
        variables: { country: "Afghanistan" }
      },
      result: { data: { Reviews: { Name: "Trond", ReviewText: "Flott land Ã¥ feriere i." , Date: "2022-10-26T13:50:35.783Z", Rating: 3.5 }}}  
    }
  ];