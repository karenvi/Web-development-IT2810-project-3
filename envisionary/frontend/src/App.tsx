import './App.css';
import { HashRouter as Router, Routes, Route } from 'react-router-dom';
import Header from './components/Header/Header';
import Feed from './pages/Feed'
import Country from './pages/Country'
import GiveReview from './pages/GiveReview';
import InfoPage from './pages/InfoPage';


function App() {
  return (
    <div className="App">
      <Router>
        <Header />
        <Routes>
          <Route path='/' element={<Feed />} />
          <Route path='/country' element={<Country />} />
          <Route path='/give-review' element={<GiveReview />} />
          <Route path='/info-page' element={<InfoPage />} />
        </Routes>
      </Router>
    </div>
  );
}

export default App;
