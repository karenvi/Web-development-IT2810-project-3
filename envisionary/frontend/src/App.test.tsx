import { render } from '@testing-library/react';
import App from './App';
import { RecoilRoot } from 'recoil';
import { MockedProvider } from "@apollo/client/testing";
import { mocks } from './__test__/mocks/mocks';



it("App renders correctly", () => {
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <RecoilRoot>
          <App/>
        </RecoilRoot>
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });

