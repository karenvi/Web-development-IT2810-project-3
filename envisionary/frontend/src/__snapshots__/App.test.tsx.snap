// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <nav
      class="header"
    >
      <div
        class="linkContainer"
      >
        <a
          class="link"
          href="/"
        >
          Countries
        </a>
        <a
          class="link"
          href="/give-review"
        >
          Give review
        </a>
      </div>
    </nav>
    <p>
      Loading...
    </p>
  </div>
</div>
`;
